cura/CuraApplication.py

```diff
@@ -235,6 +239,9 @@ def __init__(self):
 
             self._recent_files.append(QUrl.fromLocalFile(f))
 
+    def _onEngineCreated(self):
```

Comment:
Uh, why not do this in registerObjects? That gets a valid engine passed to it that you can add the image provider to.